pkg/router/template/template_helper.go

```diff
@@ -173,19 +148,34 @@ func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
 	return endpoints
 }
 
-// sortedMapData returns the sorted data in a haproxy map. The returned data is
-// in alphabetically reverse order. The sortSubGroups option sorts the
-// non-wildcard paths first and adds the sorted wildcard paths at the end.
-// Note: The reversed sorting ensures we do a longest path match first.
-func sortedMapData(name string, sortSubGroups bool) []string {
-	lines, err := fileutil.ReadLines(name)
-	if err != nil {
-		glog.Errorf("Error reading map file %s: %v", name, err)
-		return []string{}
+// backendConfig returns a haproxy backend config for a given service alias.
+func backendConfig(name string, cfg ServiceAliasConfig, hascert bool) *haproxyutil.BackendConfig {
+	return &haproxyutil.BackendConfig{
+		Name:           name,
+		Host:           cfg.Host,
+		Path:           cfg.Path,
+		IsWildcard:     cfg.IsWildcard,
+		Termination:    cfg.TLSTermination,
+		InsecurePolicy: cfg.InsecureEdgeTerminationPolicy,
+		HasCertificate: hascert,
 	}
+}
 
-	if sortSubGroups {
-		return templateutil.SortMapPaths(lines, `^[^\.]*\.`)
+// generateHAProxyCertConfigMap generates haproxy certificate config map contents.
+func generateHAProxyCertConfigMap(td templateData) []string {
+	lines := make([]string, 0)
+	for k, cfg := range td.State {
+		hascert := false
+		if len(cfg.Host) > 0 {
+			cert, ok := cfg.Certificates[cfg.Host]
+			hascert = ok && len(cert.Contents) > 0
+		}
+
+		backendConfig := backendConfig(k, cfg, hascert)
+		if entry := haproxyutil.GenerateMapEntry(certConfigMap, backendConfig); entry != nil {
```

Comment:
It would be better if this was passed in as a closure from the template name, rather than it coming from a hardcoded function.